src/Symfony/Component/Serializer/Normalizer/AbstractObjectNormalizer.php

```diff
@@ -334,7 +334,7 @@ public function denormalize(mixed $data, string $type, string $format = null, ar
             $normalizedData = $this->removeNestedValue($serializedPath->getElements(), $normalizedData);
         }
 
-        $normalizedData = array_merge($normalizedData, $nestedData);
+        $normalizedData = array_replace($normalizedData, $nestedData);
```

Comment:
Maybe I'm wrong, but I think that the array union operator (`$normalizedData = $nestedData + $normalizedData`) might be more appropriate here.

Indeed, when the `+` operator finds duplicates, it moves on instead of replacing the value, which is faster.
Moreover, it doesn't imply a function call (which is somehow expensive).

Reply:
I have never used the union operator. Thanks for your input.

I have changed it to use the union operator and all of the unit tests are running successfully. I have looked at the unittest with nestedData in detail and everything looks fine to me. I wasn't sure if the union operator would have any side effects.  

In my first commit, I did not have a unit test for this case. I added a unittest.